seed/seed16.go

```diff
@@ -191,6 +193,10 @@ func (s *seed16) addSnap(sn *internal.Snap16, essType snap.Type, pinnedTrack str
 			// TODO: consider whether to use this if we have links?
 			sideInfo.EditedContact = sn.Contact
 		}
+		if newPath != "" {
+			path = newPath
```

Comment:
Just to make sure that this is intended: a few lines below you have
```
cache[path] = seedSnap
```
and with the assignment that you make here (`path = newPath`) you are changing the cache key. Which might be correct, but at the beginning of this function we have a `seedSnap := cache[path]`, where `path` is still the one passed by the caller.

Or we should not overwrite the value of `path` here, and add both keys to the cache?
```
cache[path] = seedSnap
cache[newPath] = seedSnap
```

 Maybe this is all correct, just asking.